src/main/java/org/jnode/fs/xfs/directory/ShortFormDirectoryEntry.java

```diff
@@ -74,7 +75,7 @@ public long getINumber() {
      * @return the entry length.
      */
     public int getLength() {
-        return getNameLength() + 0x7;
+        return getNameLength() + 0x8;
```

Comment:
from reading the docs, it would seem like we have name (length) followed by 1 byte - ftype, then inumber, which can be 4 or 8 bytes. Do you have any confirmation that the inumber value is a fixed size of 8 bytes? Maybe in the test data?
... actually, even based on the `getINumber` function above, this should be dependant on the `inodeSize` value.

Reply:
yes , based in the version of testing that is v5, the value is 8.  Using a version v4 the expected value should be 4.

Reply:
```suggestion
        return getNameLength() + inodeSize;
```
That should work for v4 and v5 then?

Reply:
Hi, Erwin , yes with that change will work for both versions.

Reply:
ticket created https://github.com/Nuix/jnode-fs/projects/1#card-76547654